### ISSUE TITLE
Chart legends overlap the graph

### ISSUE THREAD

**Issue by ZeRego** [lightdash/lightdash#541]
When we have a lot of legends it may overlap the graph. 

![Screenshot 2021-10-04 at 17 44 38](https://user-images.githubusercontent.com/9117144/135892129-4ddd384a-e83d-461b-8c9d-854ad2602f6c.png)

Reading from the echarts threads [[0]](https://github.com/apache/echarts/issues/7887) [[1]](https://stackoverflow.com/questions/45452492/baidu-echarts2-dynamic-chart-position-based-on-the-legend-height), they recommend to use `type: 'scroll'`. But isn't ideal since only a few legends are readable and aren't present when we export as an image.

There is this [open issue](https://github.com/apache/echarts/issues/14127) that would make the use of 'scroll' more user friendly since we could display more rows of legends.

A possible solution is for us to somehow use the echart object ( via react ref ) and recalculate the legends height and graph height.

**Comment by hamzahc1** [comment]
This is still a relevant bug - I hate when this happens! 

Also I'm not sure if this is related but sometimes the chart axis labels also overlaps with itself: 
![Screenshot 2022-02-11 at 15 26 42](https://user-images.githubusercontent.com/14341285/153609296-3362ff04-ed3b-4be0-a43a-51a70f5796fb.png)

**Comment by ZeRego** [comment]
There are [multiple open issues in the echarts library](https://github.com/apache/echarts/issues/15654#issuecomment-1157807935) around overlapping elements. 
There are some hacks we can implement but they could stop working at any time since they rely on accessing some of their internal code/data. 

Hack we can introduce in `packages/frontend/src/components/SimpleChart/index.tsx` 
```
    useEffect(() => {
        if (eChartsOptions) {
            const echartsInstance = chartRef.current?.getEchartsInstance();
            if (echartsInstance) {
                // @ts-ignore
                const found = echartsInstance._componentsViews.find(
                    (entry: any) => entry.type === 'legend.plain',
                );
                const legendHeight = found._backgroundEl.shape.height;
                if (legendHeight) {
                    echartsInstance.setOption({
                        grid: {
                            top: legendHeight + 20,
                        },
                    });
                }
            }
        }
    }, [chartRef, eChartsOptions]);
```

We still need to make a decision on how the UI should adapt since the chart section is set to 300px so having a bigger legend would just give less height for the chart grid. 

Example:

![Screenshot 2022-06-16 at 16 22 58](https://user-images.githubusercontent.com/9117144/174110026-36dc8e6d-d11c-42f0-bfe0-3e659e603389.png)

@hamzahc1 as for the axis labels, it could be fixed under this ticket https://github.com/lightdash/lightdash/issues/2028 which will likely also need some hack.

**Comment by TuringLovesDeathMetal** [comment]
I'm going to say that this is solved by #2604 

I think there's a separate issue around _**resizing**_ the chart if your legend gets too big, but I'll wait until someone requests it to open an issue
